shared-libs/contacts/src/places.js

```diff
@@ -89,30 +83,45 @@ const validatePlace = place => {
   return Promise.resolve();
 };
 
-const createPlace = async (place) => {
-  if (place.contact && !place.contact.type) {
-    place.contact.type = people._getDefaultPersonType();
+const preparePlaceContact = async contact => {
+  if (!contact) {
+    return {};
   }
-  await module.exports._validatePlace(place);
+  if (_.isString(contact)) {
+    const person = await people.getOrCreatePerson(contact);
+    return { exists: true, contact: person };
+  }
+  if (!_.isObject(contact)) {
+    return err();
+  }
+  contact.type = contact.type ?? people._getDefaultPersonType();
+  const errStr = people._validatePerson(contact);
+  if (errStr) {
+    return err(errStr);
+  }
+  return { exists: false, contact: contact };
+};
 
-  const contact = place.contact;
+const createPlace = async (place) => {
+  await module.exports._validatePlace(place);
+  const { exists, contact } = await preparePlaceContact(place.contact);
```

Comment:
I think `exists` becomes ambiguous in this context. Maybe:
```suggestion
  const { exists: contactExists, contact } = await preparePlaceContact(place.contact);
```